File: product_api/src/main/java/com/jdc/product/api/entity/Product.java
package com.jdc.product.api.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import lombok.Data;

@Data
@Entity
public class Product {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String name;
	@ManyToOne
	private Category category;
	private int price;
	private boolean soledOut;
}

File: RestaurantPOS/app/src/main/java/com/jdc/restaurant/db/entity/Product.kt
package com.jdc.restaurant.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity
data class Product(
    @PrimaryKey(autoGenerate = true)
    var id:Long = 0,
    @ForeignKey(
        entity = Category::class,
        parentColumns = ["id"],
        childColumns = ["categoryId"],
        onDelete = ForeignKey.CASCADE
    )
    @ColumnInfo(index = true)
    var categoryId:Long = 0,
    var name:String = "",
    var price:Int = 0,
    var description:String = ""
)